### DISCUSSION TITLE
fan_level=1 RPM differ depending on system state - works correctly after waking up from sleep.

### DISCUSSION THREAD

**Discussion by al1357** [vmatare/thinkfan#144] [Support]
Hi,

thanks for nice app, really useful. I have one problem with varying fan speed.

After reboot and `systemctl stop thinkfan`, then `sudo thinkfan -n` fan works at ~2000 RPM and fan_level=1. It's the same if I don't stop it, when it runs in the background.
```
:/proc/acpi/ibm$ cat fan
status:		enabled
speed:		2067
level:		1
(...)
```

After going into and waking up from sleep `sudo thinkfan -n` runs it at ~1600 RPM and also shows fan_level=1
```
:/proc/acpi/ibm$ cat fan
status:		enabled
speed:		1608
level:		1
(...)
```

1600 RPM is correct - it runs like that on Windows after installing all Lenovo drivers. I have TP T540p.

My `/etc/thinkfan.conf`:
```
hwmon /sys/devices/platform/coretemp.0/hwmon/hwmon4/temp3_input
hwmon /sys/devices/platform/coretemp.0/hwmon/hwmon4/temp4_input
hwmon /sys/devices/platform/coretemp.0/hwmon/hwmon4/temp1_input
hwmon /sys/devices/platform/coretemp.0/hwmon/hwmon4/temp5_input
hwmon /sys/devices/platform/coretemp.0/hwmon/hwmon4/temp2_input

(0,	0,	40)
(1,	35,	60)
(2,	50,	61)
(3,	52,	63)
(4,	56,	65)
(5,	59,	66)
(7,	63,	79)
(127,	75,	32767)
```

What can I do to make it run 1600RPM on level 1 after restart, without going into sleep?

**Comment by vmatare** [comment]
Thinkfan has no influence on what your embedded controller and fan ultimately do with the `level 1` given to them. Sounds to me like it could be an issue with the `thinkpad_acpi` kernel module (or at least its developers might be better able to help you), or something is wrong with your embedded controller. The first thing I'd try is to do a BIOS update. It usually comes with an update to the embedded controller firmware as well, and that's the part that is ultimately responsible for mapping some `level 1` to fan RPM.
